Source/WebCore/rendering/style/StyleGradientImage.cpp

```diff
@@ -614,9 +614,13 @@ GradientColorStops StyleGradientImage::computeStops(GradientAdapter& gradientAda
                     stops.append(newStop);
                     if (currOffset > maxExtent)
                         break;
-                    if (srcStopOrdinal < originalNumberOfStops - 1)
+
+                    if (++srcStopOrdinal < originalNumberOfStops) {
                         currOffset += *stops[srcStopIndex + 1].offset - *stops[srcStopIndex].offset;
-                    srcStopOrdinal = (srcStopOrdinal + 1) % originalNumberOfStops;
+                        continue;
+                    }
+
+                    srcStopOrdinal %= originalNumberOfStops;
```

Comment:
Your new code doesn't look equivalent to the original code. Can you please explain your changes and prove that they are correct?

Reply:
If srcStopOrdinal < originalNumberOfStops, then srcStopOrdinal % originalNumberOfStops is always srcStopOrdinal